state/protocol/protocol_state/kvstore/models.go

```diff
@@ -9,22 +9,31 @@ import (
 	"github.com/onflow/flow-go/state/protocol/protocol_state"
 )
 
-type upgradableModel struct {
+// This file contains the concrete types that define the structure of the underlying key-value store
+// for a particular Protocol State version.
+// Essentially enumerating the set of keys and values that are supported.
+// When a key is added or removed, this requires a new protocol state version.
+// To use new version of the protocol state, create a new versioned model in models.go (eg. modelv3 if latest model is modelv2)
+// ATTENTION: All models should be public with public fields otherwise the encoding/decoding will not work.
+
+// UpgradableModel is a utility struct that can be embedded in any model to provide
+// a common interface for managing protocol version upgrades.
+type UpgradableModel struct {
 	VersionUpgrade *protocol_state.ViewBasedActivator[uint64]
```

Comment:
not sure about this, the whole reason for it to be a pointer is to be an optional value. We don't always have `VersionUpgrade` available, one example is spinning up a chain so a pointer made a lot of sense to me, especially to avoid introducing some initial state to the `ViewBasedActivator[uint64]`